torch/csrc/stable/tensor_struct.h

```diff
@@ -74,12 +74,30 @@ class Tensor {
   // semantics as their counterparts in TensorBase.h.
   // =============================================================================
 
+  // Do not add new uses of data_ptr(), use const_data_ptr() if
+  // possible, mutable_data_ptr() otherwise.
   void* data_ptr() const {
     void* data_ptr;
     TORCH_ERROR_CODE_CHECK(aoti_torch_get_data_ptr(ath_.get(), &data_ptr));
     return data_ptr;
   }
 
+  template <typename T>
+  T* mutable_data_ptr() const {
```

Comment:
What's the difference between having the template version be on the implementation directly versus what we have in TensorBase.h:

```
  const void* const_data_ptr() const {
    return this->unsafeGetTensorImpl()->data();
  }

  void* mutable_data_ptr() const {
    return this->unsafeGetTensorImpl()->mutable_data();
  }
...
template <typename T, std::enable_if_t<!std::is_const_v<T>, int> = 0>
  const T* const_data_ptr() const;

  template <typename T, std::enable_if_t<std::is_const_v<T>, int> = 0>
  const std::remove_const_t<T>* const_data_ptr() const;

  template <typename T>
  T* mutable_data_ptr() const;
```
?

Reply:
This PR now exposes non-template methods `mutable_data_ptr` and `const_data_ptr`. The corresponding template methods (with proper type checking) will be implemented in a follow-up.

Reply:
I think we should add the templating in the same PR to match the semantics of the nonstable version.

Reply:
> I think we should add the templating in the same PR to match the semantics of the nonstable version.

This is now done in this PR.